test/t/001_extension.pl

```diff
@@ -30,23 +30,13 @@ sub check_extension_state
 	# report test failures from caller location
 	local $Test::Builder::Level = $Test::Builder::Level + 1;
 
-	# reset output collector
-	$out = "";
-	# restart per-command timer
-	$timer->start(5);
-	# send the data to be sent
-	$in .= "SELECT * FROM extension_state();\n";
-	# wait ...
+	my $result = $node->safe_psql(
+		'postgres',
+		q{
+		SELECT * FROM extension_state();
+	});
 
-	pump $h until ($out =~ $pattern || $timer->is_expired);
-	my $okay = ($out =~ $pattern && !$timer->is_expired);
-	ok($okay, $annotation);
-	# for debugging, log actual output if it didn't match
-	local $Data::Dumper::Terse = 1;
-	local $Data::Dumper::Useqq = 1;
-	diag 'Actual output was ' . Dumper($out) . "Did not match \"$pattern\"\n"
-	  if !$okay;
-	return;
+	is($result, $pattern, $annotation);
```

Comment:
if I understand correctly, we no longer do any waiting, if that is the case, shouldn't we make sure we wait for the extension state to change? 
Maybe with something similar to poll_query_until?

Reply:
Timeouts are not required here as this is a very simple function call that should return immediately. We still wait for the response from the server.

https://github.com/postgres/postgres/blob/master/src/test/perl/README